exporter/collector/monitoredresource.go

```diff
@@ -0,0 +1,166 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"go.opentelemetry.io/collector/model/pdata"
+	semconv "go.opentelemetry.io/collector/model/semconv/v1.8.0"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+)
+
+const (
+	awsAccount     = "aws_account"
+	awsEc2Instance = "aws_ec2_instance"
+	clusterName    = "cluster_name"
+	containerName  = "container_name"
+	gceInstance    = "gce_instance"
+	genericNode    = "generic_node"
+	genericTask    = "generic_task"
+	instanceID     = "instance_id"
+	job            = "job"
+	k8sCluster     = "k8s_cluster"
+	k8sContainer   = "k8s_container"
+	k8sNode        = "k8s_node"
+	k8sPod         = "k8s_pod"
+	location       = "location"
+	namespace      = "namespace"
+	namespaceName  = "namespace_name"
+	nodeID         = "node_id"
+	nodeName       = "node_name"
+	podName        = "pod_name"
+	region         = "region"
+	taskID         = "task_id"
+)
+
+var (
+	// monitoredResourceMappings contains mappings of GCM resource label keys onto OTel
+	// resource keys for a given monitored resource type. For entries with multiple OTel
+	// resource keys, the keys' values will be coalesced in order until there is a non-empty
+	// value.
+	monitoredResourceMappings = map[string]map[string][]string{
+		gceInstance: {
+			location:   {semconv.AttributeCloudAvailabilityZone},
+			instanceID: {semconv.AttributeHostID},
+		},
+		k8sContainer: {
+			location:      {semconv.AttributeCloudAvailabilityZone},
+			clusterName:   {semconv.AttributeK8SClusterName},
+			namespaceName: {semconv.AttributeK8SNamespaceName},
+			podName:       {semconv.AttributeK8SPodName},
+			containerName: {semconv.AttributeK8SContainerName},
+		},
+		k8sPod: {
+			location:      {semconv.AttributeCloudAvailabilityZone},
+			clusterName:   {semconv.AttributeK8SClusterName},
+			namespaceName: {semconv.AttributeK8SNamespaceName},
+			podName:       {semconv.AttributeK8SPodName},
+		},
+		k8sNode: {
+			location:    {semconv.AttributeCloudAvailabilityZone},
+			clusterName: {semconv.AttributeK8SClusterName},
+			nodeName:    {semconv.AttributeK8SNodeName},
+		},
+		k8sCluster: {
+			location:    {semconv.AttributeCloudAvailabilityZone},
+			clusterName: {semconv.AttributeK8SClusterName},
+		},
+		awsEc2Instance: {
+			instanceID: {semconv.AttributeHostID},
+			region:     {semconv.AttributeCloudAvailabilityZone},
+			awsAccount: {semconv.AttributeCloudAccountID},
+		},
+		genericTask: {
+			location:  {semconv.AttributeCloudAvailabilityZone, semconv.AttributeCloudRegion},
+			namespace: {semconv.AttributeServiceNamespace},
+			job:       {semconv.AttributeServiceName},
+			taskID:    {semconv.AttributeServiceInstanceID},
+		},
+		genericNode: {
+			location:  {semconv.AttributeCloudAvailabilityZone, semconv.AttributeCloudRegion},
```

Comment:
The spec says that on the last fall through, the location should be populated with "global" if semantic convention labels aren't populated. This is because writing with a location set to empty string will be an error. Do you think we need to handle that case?

Reply:
I think we *can't* use global if we default to workload.googleapis.com right?  DO we need to update the spec?

Reply:
Fixed for `generic_task/generic_node`. For the other resources it should really be present, so I'll leave those to empty to surface the error

Reply:
Setting the value of the "location" label to "global" is still allowed under workload.googleapis.com.